gradio/utils.py

```diff
@@ -1057,13 +1057,10 @@ def is_in_or_equal(path_1: str | Path, path_2: str | Path) -> bool:
         path_1: str or Path (to file or directory)
         path_2: str or Path (to file or directory)
     """
-    path_1, path_2 = (
-        os.path.normpath(os.path.abspath(path_1)),
-        os.path.normpath(os.path.abspath(path_2)),
-    )
+    path_1, path_2 = abspath(path_1).resolve(), abspath(path_2).resolve()
```

Comment:
We reverted back to abspath() and resolve()? I thought we had changed from this to os.path.normpath(os.path.abspath(path_1)) previously

Reply:
Originally it was just `abspath`. In #9020 I first added the resolve but then refactored to the implementation we have here to avoid the symlink resolution. But I think we need the `resolve` to handle all the corner cases around directory names. Since the symlink resolution is just being done within this check, I think it's ok.

Reply:
But yes we've been iterating here 😅

Reply:
makes sense 👍